spyder/plugins/onlinehelp/widgets.py

```diff
@@ -322,27 +326,31 @@ def initialize(self, checked=True):
             This method is connected to the `sig_toggle_view_changed` signal,
             so that the first time the widget is made visible it will start
             the server. Default is True.
+        force: bool, optional
+            Force a server start even if the server is running.
+            Default is False.
         """
-        if checked and self.server is None:
+        if force or (checked and self.server is None
+                     or not self.is_server_running()):

```

Comment:
```suggestion
        if (
            force or
            (checked and self.server is None or
             not self.is_server_running())
         ):
```

Style improvement.

Reply:
I haven't looked at all of this in detail, but I'm really excited about what you've been doing @dalthviz - thanks so much! One small comment on this: while @ccordoba12's suggestion is an improvement, it's still a somewhat opaque complex condition.

The first simplification is that the pair of parentheses is unnecessary: the current structure is `A or (B and C or D)`, which is equivalent to `A or B and C or D`; clearer, though, would be `A or (B and C) or D`.

The second is that the condition `checked and self.server is None` could be assigned to a variable for clarity.  For example (and this is obviously not correct, but it should illustrate what I mean):

```python
        no_server_needed = checked and self.server is None
        if force or no_server_needed or not self.is_server_running():
            ...
```
In this way, the condition comfortably sits on one line and is much easier to read.